### ISSUE TITLE
Cannot install syndesis from master - roles.rbac.authorization.k8s.io "syndesis-operator" not found

### ISSUE THREAD

**Issue by paoloantinori** [syndesisio/syndesis#3270]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem

Current master, Aug 2nd 2018, fails to deploy due to Operators related errors

## Tasks involved / Steps to Reproduce

1. ` ./syndesis install -p myproject --verbose`
2.
3.
4.

Error:

```Error from server (NotFound): error when creating "https://raw.githubusercontent.com/syndesisio/syndesis/master/install/operator/deploy/syndesis-operator.yml": roles.rbac.authorization.k8s.io "syndesis-operator" not found
++(/data/repositories/work/syndesis/syndesis/tools/bin/commands/install:59): install::run(): result='serviceaccount "syndesis-operator" created
rolebinding "syndesis-operator:view" created
rolebinding "syndesis-operator:edit" created
imagestream "syndesis-operator" created
deploymentconfig "syndesis-operator" created'


```

Relates to #3260 despite this one reports a failure at pure install level, while the other is about building.

**Comment by paoloantinori** [comment]
Workaround from IRC:

```
 logged in as admin, call: syndesis install --setup-crd and for minishift: syndesis install --fix-permissions developer Alternative: Use "sy minishift" which does all of this in one step.
```

**Comment by rhuss** [comment]
@paoloantinori 

Tried `syndesis install --help` ?

**Comment by paoloantinori** [comment]
not yet. just used the command that used to work and that fails now with a technical error instead of a user level one.

**Comment by rhuss** [comment]
Ok, I will improve the error message. thanks for the feedback.

**Comment by rhuss** [comment]
Still, "syndesis" is a developer tool, not meant for endusers. The installation script for enduser on OCP (and syndesis) is still missing which I will start when I have fixed all the p0 bugs around syndesis.

**Comment by paoloantinori** [comment]
thank you! by the way, I'm testing this on OS 3.10 so I need to approach this in a "brain less" way to be able to distinguish eventual issues with our scripts compared to possible limits/change coming from OS 3.10

**Comment by rhuss** [comment]
come on, we are pros with a brain, aren't we ;-) But happy to arrange the priorities that way.

I will fix the developer doc for sure, just not sure whether it should be the first priority.

**Comment by paoloantinori** [comment]
well know that the workaround has been confirmed and documented we can downgrade it, despite I still find this being a top priority. Anyhow, if it's just a matter of adding the steps I can do that. If you want to spend some word on why this is required, to give context, I'll leave it to you.

**Comment by heiko-braun** [comment]
@rhuss For the records: What's "the workaround from IRC"?

**Comment by paoloantinori** [comment]
And yes, you are right. 

Normallly if a workaround/fix exists that makes an issue a non-blocker, my bad.

**Comment by rhuss** [comment]
Damned, not a workaround ! That is the final solution ...

**Comment by heiko-braun** [comment]
ah, i see it

**Comment by zregvart** [comment]
> come on, we are pros with a brain, aren't we

speak for yourself, I'm happy being professionally brain dead

![giphy](https://user-images.githubusercontent.com/1306050/43569415-b0cf5022-9637-11e8-97e6-60430a5e00a2.gif)

**Comment by oscerd** [comment]
Just for docs I've done in this way and it looks like it works.

```
oscerd@ghost:~/workspace/jboss-fuse/syndesis$ minishift start --vm-driver virtualbox --memory 6128
-- Starting profile 'minishift'
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.9.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.9.0' is supported ... OK
-- Checking if requested hypervisor 'virtualbox' is supported on this platform ... OK
-- Checking if VirtualBox is installed ... OK
-- Checking the ISO URL ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting local OpenShift cluster using 'virtualbox' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    5 GB
   vCPUs :    2
   Disk size: 20 GB
-- Starting Minishift VM .......................... OK
-- Checking for IP address ... OK
-- Checking for nameservers ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 0% used OK
   Importing 'openshift/origin:v3.9.0' ............ OK
   Importing 'openshift/origin-docker-registry:v3.9.0' ...... OK
   Importing 'openshift/origin-haproxy-router:v3.9.0' .. OK
-- OpenShift cluster will be configured with ...
   Version: v3.9.0
Using Docker shared volumes for OpenShift volumes
Using 192.168.99.100 as the server IP
Starting OpenShift using openshift/origin:v3.9.0 ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.99.100:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

oscerd@ghost:~/workspace/jboss-fuse/syndesis$ oc adm policy  --as system:admin add-cluster-role-to-user cluster-admin developer
cluster role "cluster-admin" added: "developer"
oscerd@ghost:~/workspace/jboss-fuse/syndesis$ tools/bin/syndesis install -s
Installing Syndesis CRD
oscerd@ghost:~/workspace/jboss-fuse/syndesis$ tools/bin/syndesis minishift -p test --install
Switched to context "minishift".
Creating project test
Already on project "test" on server "https://192.168.99.100:8443".
role "syndesis-extra-permissions" created
role "syndesis-extra-permissions" added: "developer"
Login successful.

You have access to the following projects and can switch between them with 'oc project <projectname>':

    default
    kube-public
    kube-system
    myproject
    openshift
    openshift-infra
    openshift-node
    openshift-web-console
  * test

Using project "test".
Waiting for syndesis-operator to be scaled to 1
Sleeping 10s ...
NAME                         READY     STATUS    RESTARTS   AGE
syndesis-operator-1-deploy   0/1       Pending   0          0s
syndesis-operator-1-deploy   0/1       Pending   0         0s
syndesis-operator-1-deploy   0/1       ContainerCreating   0         0s
syndesis-operator-1-lr579   0/1       Pending   0         0s
syndesis-operator-1-lr579   0/1       Pending   0         0s
syndesis-operator-1-lr579   0/1       ContainerCreating   0         0s
syndesis-operator-1-deploy   1/1       Running   0         1s
Sleeping 10s ...
syndesis-operator-1-lr579   1/1       Running   0         14s
syndesis-operator-1-deploy   0/1       Completed   0         16s
syndesis-operator-1-deploy   0/1       Terminating   0         16s
syndesis-operator-1-deploy   0/1       Terminating   0         16s
Waiting for syndesis-server to be scaled to 1
Sleeping 10s ...
NAME                        READY     STATUS    RESTARTS   AGE
syndesis-operator-1-lr579   1/1       Running   0          17s
syndesis-db-1-deploy   0/1       Pending   0         0s
syndesis-db-1-deploy   0/1       Pending   0         0s
syndesis-db-1-deploy   0/1       ContainerCreating   0         0s
syndesis-oauthproxy-1-deploy   0/1       Pending   0         0s
syndesis-oauthproxy-1-deploy   0/1       Pending   0         0s
syndesis-oauthproxy-1-deploy   0/1       ContainerCreating   0         0s
syndesis-ui-1-deploy   0/1       Pending   0         0s
syndesis-ui-1-deploy   0/1       Pending   0         0s
syndesis-ui-1-deploy   0/1       ContainerCreating   0         0s
syndesis-meta-1-deploy   0/1       Pending   0         0s
syndesis-meta-1-deploy   0/1       Pending   0         0s
syndesis-meta-1-deploy   0/1       ContainerCreating   0         0s
todo-1-build   0/1       Pending   0         0s
todo-1-build   0/1       Pending   0         0s
todo-1-build   0/1       Init:0/2   0         0s
syndesis-db-1-gcg2r   0/1       Pending   0         0s
syndesis-db-1-gcg2r   0/1       Pending   0         0s
syndesis-server-1-deploy   0/1       Pending   0         0s
syndesis-server-1-deploy   0/1       Pending   0         0s
syndesis-db-1-gcg2r   0/1       ContainerCreating   0         0s
syndesis-prometheus-1-deploy   0/1       Pending   0         0s
syndesis-prometheus-1-deploy   0/1       Pending   0         0s
syndesis-oauthproxy-1-7ccrc   0/1       Pending   0         0s
syndesis-oauthproxy-1-7ccrc   0/1       Pending   0         0s
syndesis-server-1-deploy   0/1       ContainerCreating   0         1s
syndesis-meta-1-fszrr   0/1       Pending   0         0s
syndesis-meta-1-fszrr   0/1       Pending   0         0s
syndesis-ui-1-jn42d   0/1       Pending   0         0s
syndesis-ui-1-jn42d   0/1       Pending   0         0s
syndesis-prometheus-1-deploy   0/1       ContainerCreating   0         3s
syndesis-prometheus-1-f6bf7   0/1       Pending   0         0s
syndesis-prometheus-1-f6bf7   0/1       Pending   0         0s
syndesis-server-1-hbklc   0/1       Pending   0         0s
syndesis-server-1-hbklc   0/1       Pending   0         0s
syndesis-oauthproxy-1-7ccrc   0/1       ContainerCreating   0         5s
Sleeping 10s ...
syndesis-db-1-deploy   1/1       Running   0         9s
syndesis-oauthproxy-1-deploy   1/1       Running   0         9s
syndesis-server-1-hbklc   0/1       ContainerCreating   0         4s
syndesis-meta-1-deploy   1/1       Running   0         10s
syndesis-prometheus-1-deploy   1/1       Running   0         9s
syndesis-meta-1-fszrr   0/1       ContainerCreating   0         9s
syndesis-ui-1-jn42d   0/1       ContainerCreating   0         9s
syndesis-prometheus-1-f6bf7   0/1       ContainerCreating   0         7s
syndesis-ui-1-deploy   1/1       Running   0         12s
syndesis-server-1-deploy   1/1       Running   0         12s
todo-1-build   0/1       Init:0/2   0         13s
todo-1-build   0/1       Init:1/2   0         14s
todo-1-build   0/1       PodInitializing   0         15s
todo-1-build   1/1       Running   0         16s
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
syndesis-db-1-gcg2r   0/1       Running   0         2m
syndesis-db-1-gcg2r   1/1       Running   0         2m
syndesis-db-1-deploy   0/1       Completed   0         2m
syndesis-db-1-deploy   0/1       Terminating   0         2m
syndesis-db-1-deploy   0/1       Terminating   0         2m
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
Sleeping 10s ...
syndesis-oauthproxy-1-7ccrc   0/1       Running   0         3m
Sleeping 10s ...
```

**Comment by rhuss** [comment]
Thanks !

> oscerd@ghost:~/workspace/jboss-fuse/syndesis$ oc adm policy --as system:admin add-cluster-role-to-user cluster-admin developer cluster role "cluster-admin" added: "developer"

Actually with minishift it should also work with 

```
syndesis install --fix-permissions developer
```

instead of attaching full cluster admin to developer. If not, please let me know.
FYI, https://github.com/syndesisio/syndesis/pull/3213#issuecomment-408915960 summarize the changes. Big apologies that I forgot to move that info around. To much in my head these days.

**Comment by oscerd** [comment]
Ok, I'll test and get back to you here.

**Comment by oscerd** [comment]
Thanks!

**Comment by oscerd** [comment]
syndesis install --fix-permissions developer 

will work only if you already setup the CRD, but to setup CRD you'll need to be cluster-admin, so the command I posted is the only working for me.

**Comment by rhuss** [comment]
good point. You would need also cluster admin for --fix-permissions and you are right --setup-crd needs to be performed before.

Alternatively you can do what `syndesis minishift` does:

```
minishift addons enable admin-user
oc login -u system:admin
syndesis install --setup-crd
syndesis install --fix-permissions developer
oc login -u developer
syndesis install 
```

I will take your suggestion and combine, `setup-crd` and `fix-permissions` to a `setup-as-admin` option. 

Thanks for the feedback !

**Comment by rhuss** [comment]
fixed with #3279
